cassiopeia/plotting/itol_utilities.py

```diff
@@ -0,0 +1,623 @@
+
+            SIDout.write("\nLEGEND_LABELS")
+            for key in colormap.keys():
+                SIDout.write(f"\t{key}")
+            SIDout.write("\n")
+
+        SIDout.write("\nDATA\n")
+        df_writeout = dfCellColor.to_csv(
+            None, sep="\t", header=False, index=False
+        )
+        SIDout.write(df_writeout)
+
+    return outfp
+
+
+def create_indel_heatmap(
+    alleletable: pd.DataFrame,
+    tree: CassiopeiaTree,
+    dataset_name: str,
+    output_directory: str,
+    indel_colors: Optional[pd.DataFrame] = None,
+    indel_priors: Optional[pd.DataFrame] = None,
+    random_state: Optional[np.random.RandomState] = None,
+) -> List[str]:
+    """Creates a set of files for displaying an indel heatmap with iTOL
+
+    Creates a set of files, each one corresponding to a character, that can be
+    used to display an allele heatmap alongside a tree in iTOL. If neither indel
+    colors nor indel priors are provided, a random color mapping is created for
+    each unique indel.
+
+    Args:
+        alleletable: An AlleleTable for the tree
+        tree: CassiopeiaTree
+        dataset_name: Name for the dataset
+        output_directory: Where to write the output files
+        indel_colors: Mapping of indels to colors.
+        indel_priors: Prior probabilities for each indel. Only `indel_colors`
+            are not provided, in which case a new indel color mapping is created
+            by displaying low-probability indels with bright colors and
+            high-probability ones with dull colors.
+        random_state: Random state for reproducibility
+
+    Returns:
+        A set of filepaths to each cut-site's color bar file for iTOL batch
+            uploading.
+    """
+
+    _leaves = tree.leaves
+
+    lineage_profile = utilities.convert_alleletable_to_lineage_profile(
+        alleletable
+    )
+    clustered_linprof = lineage_profile.loc[_leaves[::-1]]
+
+    if indel_colors is None:
+        if indel_priors is None:
+            indel_colors = get_random_indel_colors(lineage_profile, random_state)
+        else:
+            indel_colors = get_indel_colors(indel_priors, random_state)
+
+    # Convert colors and make colored alleleTable (rgb_heatmap)
+    r, g, b = (
+        np.zeros(clustered_linprof.shape),
+        np.zeros(clustered_linprof.shape),
+        np.zeros(clustered_linprof.shape),
+    )
+    for i in tqdm(range(clustered_linprof.shape[0])):
+        for j in range(clustered_linprof.shape[1]):
+            ind = str(clustered_linprof.iloc[i, j])
+            if ind == "nan":
+                r[i, j], g[i, j], b[i, j] = 1, 1, 1
+            elif "none" in ind.lower():
+                r[i, j], g[i, j], b[i, j] = 192 / 255, 192 / 255, 192 / 255
+            else:
+                col = hsv_to_rgb(tuple(indel_colors.loc[ind, "color"]))
+                r[i, j], g[i, j], b[i, j] = col[0], col[1], col[2]
+
+    rgb_heatmap = np.stack((r, g, b), axis=2)
+
+    alfiles = []
```

Comment:
What does `alfiles` mean? Is it "all files"?

Reply:
No, these are "allele files". I'll clarify this.